### ISSUE TITLE
[CoE Starter Kit - BUG] ISSUE TITLE

### ISSUE THREAD

**Issue by Dyson-PC** [microsoft/coe-starter-kit#6917]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

This is a brand new deployment of the CoE toolkit Core Components, with BYODL. The Environments Dataflow fails at the 'Replaced SYSTEM with ID' step, with the following message:

Expression.Error: There weren't enough elements in the enumeration to complete the operation.DetailsReason = Expression.Error
Detail = #table({"admin_makerisorphaned", "admin_userprincipalname", "admin_recordguidasstring", "admin_makerid", "admin_managerdisplayname", "admin_useremail", "admin_displayname", "admin_stateorprovince", "admin_admin_dlpimpactanalysis_ImpactOwner_admin", "admin_admin_solution_SolutionOwner_admin_maker"}, {})

I've checked the Makers table and I can't see a SYSTEM record in there - could this be the issue?

### Expected Behavior

The Dataflow to complete successfully.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.16

### What app or flow are you having the issue with?

Environments Dataflow

### What method are you using to get inventory and telemetry?

Data Export

### Steps To Reproduce

1. Installed the Core Components solution and followed the online instructions to start the Sync Template flows
2. Changed the Environment Web API and Data Storage URL in the Makers Dataflow, signed in as the service account and published the Dataflow. Waited for the Dataflow run to complete successfully
3. Re-entered the same Environment Web API and Data Storage URL in the Environments Dataflow, signed in as the service account, but could not get data from the Environments query
4. Traced the issue to the Replaced SYSTEM with ID step by going through each applied step in the query settings

AB#287

**Comment by Jenefer-Monroe** [comment]
Hello have you used the Setup Wizard to setup? For BYODL we rely on you using the Setup Wizard for it to configure things like the SYSTEM user correctly.

**Comment by Dyson-PC** [comment]
Hey, thanks for the quick response. No, I went through the manual instructions. I'll remove the rows from the Maker and Environment tables and will try again, thanks! 

As I followed the manual steps, I set the Inventory and Telemetry in Azure Data Storage account environment variable to Yes first and then started all the Sync Template flows. Does it matter that flows like the Sync Template v4 (Apps) are started, even though I am using Data Export? I see these flows do a check for BYODL anyway and terminate the flow if set to Yes
